frontend/src/components/ui/question/QuestionBox.js

```diff
@@ -46,7 +48,7 @@ function QuestionBox() {
         });
     }
     fetchData();
-  }, []);
+  }, [title, categories, content, difficulty, roomId, questionNumber]);
```

Comment:
have you check if this will cause infinite loop? You can put a console.log in this component to check if it keeps rendering. Remember to remove the console.log tho after testing, thanks!

Reply:
Thanks, you're right, it goes into an infinite loop! I have removed the unneccesary ones.